transport/controller.cc

```diff
@@ -112,47 +112,57 @@ future<> controller::do_start_server() {
         std::vector<listen_cfg> configs;
         int native_port_idx = -1, native_shard_aware_port_idx = -1;
 
-        if (cfg.native_transport_port.is_set() ||
-                (!cfg.native_transport_port_ssl.is_set() && !cfg.native_transport_port.is_set())) {
-            // Non-SSL port is specified || neither SSL nor non-SSL ports are specified
-            configs.emplace_back(listen_cfg{ socket_address{ip, cfg.native_transport_port()}, false });
-            _listen_addresses.push_back(configs.back().addr);
-            native_port_idx = 0;
-        }
-        if (cfg.native_shard_aware_transport_port.is_set() ||
-                (!cfg.native_shard_aware_transport_port_ssl.is_set() && !cfg.native_shard_aware_transport_port.is_set())) {
-            configs.emplace_back(listen_cfg{ socket_address{ip, cfg.native_shard_aware_transport_port()}, true });
-            _listen_addresses.push_back(configs.back().addr);
-            native_shard_aware_port_idx = native_port_idx + 1;
-        }
-
-        // main should have made sure values are clean and neatish
-        if (utils::is_true(utils::get_or_default(ceo, "enabled", "false"))) {
-            auto cred = std::make_shared<seastar::tls::credentials_builder>();
-            utils::configure_tls_creds_builder(*cred, std::move(ceo)).get();
+        if (!_enable_maintenance_port) {
+            const seastar::net::inet_address ip = utils::resolve(cfg.rpc_address, family, preferred).get0();
```

Comment:
Why disable the regular port if the maintenance port is open?

I can understand disabling it under maintenance mode, but as a regular server we can serve both ports.

Reply:
For the maintenance port, a new cql controller is created, because it uses a separate authorizer, role manager, and authenticator and it doesn't use metrics.
Both cql controllers are enabled in a regular server.